package/MDAnalysis/core/selection.py

```diff
@@ -656,7 +657,12 @@ def _apply(self, group):
         if not pattern:
             raise ValueError(f"{self.pattern!r} is not a valid SMARTS query")
         mol = group.convert_to("RDKIT", **self.rdkit_kwargs)
-        matches = mol.GetSubstructMatches(pattern, useChirality=True)
+        # override GetSubstructMatches default values
+        if "useChirality" not in self.smarts_kwargs:
+            self.smarts_kwargs["useChirality"] = True
+        if "maxMatches" not in self.smarts_kwargs:
+            self.smarts_kwargs["maxMatches"] = group.n_atoms
```

Comment:
I think someone is going to have to convince me about setting goup.n_atoms here as the default instead of just using the standard default. Particularly in high symmetry,  low N molecules, do we run the risk of setting this too low?

It's best we be as clear and future proof as possible here so we minimise user impact.

Reply:
I don't think the default limit of 1000 matches is appropriate for MD simulations, and I doubt that's what the RDkit writes had in mind when they set it. 

I often use the smarts query to select a class of molecules by querying for a unique functional group. It's nice to do that without thinking about system size. IMO the default of 1000 leads to inconsistent behavior, where in some cases I will have no issues but when the simulations is slightly larger I will hit an arbitrary cutoff.

A default that scales with system sizes seems more future proof to me than a low cutoff.

Reply:
I'm going to have to ask for some quantitative specifics here please.

I get that for large systems this is an issue but what about small ones - does a symmetric low N case cause problems? are there other edge cases we can think of? Every time we change a default we need to 2 minor versions (indeed we can't implement this change in 2.1.0 without warning users). If we change defaults I'd also like it so that users know exactly the bounds of what they are getting themselves into. (I know this sounds harsh but please don't take it as such, it's just that folks rely on this for science so we need to be as clear as possible in our decisions)

Reply:
Ahhh, I see your point regarding low N molecules now. For a more backwards compatible default, we could use `max(1000, group.n_atoms)`?

The two downsides I can think of for a higher default are a) wasted compute for high-symmetry systems and b) it could disrupt a users workflow if they are relying on the hard cap of 1000. For a) I believe a cap of `n_atoms` would prevent too much wasted compute. For b) hopefully adequate warning would prevent any trouble. Otherwise, I can't think of any cases where increasing the selection cap would have a detrimental. I can more quantitatively benchmark a) if needed.